web/src/user_profile.ts

```diff
@@ -1129,12 +1143,20 @@ function toggle_submit_button($edit_form: JQuery): void {
     const current_values = get_current_values($edit_form);
     const $submit_button = $("#user-profile-modal .dialog_submit_button");
     const full_name_value = $edit_form.find<HTMLInputElement>("input[name='full_name']").val()!;
+    const $short_name = $edit_form.find<HTMLInputElement>("input[name='bot_short_name']");
+    const short_name_value = $short_name.val()?.trim();
 
     if (full_name_value.trim() === "") {
         $submit_button.prop("disabled", true);
         return;
     }
 
+    // Only validate short_name if we're editing a bot (field exists)
+    if (short_name_value !== undefined && short_name_value === "") {
```

Comment:
These `undefined` checks are not needed. See how we use non-null assertion operator for full names.